src/parser.rs

```diff
@@ -5482,14 +5490,25 @@ impl<'a> Parser<'a> {
         }
     }
 
-    pub fn parse_values(&mut self) -> Result<Values, ParserError> {
-        let values = self.parse_comma_separated(|parser| {
+    pub fn parse_values(&mut self, within_insert: bool) -> Result<Values, ParserError> {
+        let mut explicit_row = false;
+
+        let rows = self.parse_comma_separated(|parser| {
+            if parser.parse_keyword(Keyword::ROW) {
+                explicit_row = true;
+            } else {
+                if !within_insert && parser.dialect.values_require_row_in_select() {
```

Comment:
How important is it to reject the ROWS keyword during inserts?  I see that mySQL is not consistent in that ROW is not allowed in INSERT statements https://dev.mysql.com/doc/refman/8.0/en/insert.html

However, I also don't see any (negative) tests for this case in this PR, though maybe I have missed them

My preference is to remove the special case handling for in select and make the parser permissive and support an optional ROW whenever a VALUES list is specified.

Downstream crates that care can detect `explicit_row` and error if they don't want to allow that syntax.

We tried to make the philosophy of "parse all the things" clearer recently in the introduction here https://github.com/sqlparser-rs/sqlparser-rs#extensible-sql-lexer-and-parser-for-rust

Reply:
BTW I found

```
Quality of my code reflects banality of this detail.
```

😆

Reply:
Yeah that makes sense - there is no need to inforce explict row. And it simplifies the code a lot!